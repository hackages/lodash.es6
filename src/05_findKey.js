/**
 * This method is like _.find except that it returns the key of the first element predicate returns truthy for instead
 * of the element itself.
 */
export function findKey(object, predicate) {}
