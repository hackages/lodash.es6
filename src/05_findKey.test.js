import {findKey} from './05_findKey'

describe('findKey', () => {
  const users = {
    barney: {age: 36, active: true},
    fred: {age: 40, active: false},
    pebbles: {age: 1, active: true},
  }

  it('should work with functions', () => {
    expect(
      findKey(users, function(o) {
        return o.age < 40
      })
    ).toMatchSnapshot()
  })

  it('should work with objects', () => {
    expect(findKey(users, {age: 1, active: true})).toMatchSnapshot()
  })
})
