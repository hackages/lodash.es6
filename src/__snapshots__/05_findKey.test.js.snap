// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findKey should work with functions 1`] = `"barney"`;

exports[`findKey should work with objects 1`] = `"pebbles"`;
